### ISSUE TITLE
Deprecate the bin/tapconvert utility

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5038]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5038|
|Type|task|
|Created|2011-04-07 17:39:20Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-tapconvert-5038|

[twisted/scripts/tapconvert.py](twisted/scripts/tapconvert.py) hasn't been updated in 6 years (except for a copyright header change) and can be deprecated.

<details><summary>Searchable metadata</summary>

```
trac-id__5038 5038
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_deprecate_tapconvert_5038 branches/deprecate-tapconvert-5038
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1302197960000000 1302197960000000
changetime__1329683946000000 1329683946000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [33520]) Branching to 'deprecate-tapconvert-5038'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33522]) deprecate tapconvert, refs [#5038](https://github.com/twisted/twisted/issues/5038)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for all your work cleaning up the cruft parts of the Twisted code base, thijs.

There's a copy-paste error from `test_deprecatedTkunzip` to `test_deprecatedTapconvert` that causes the new deprecation test to fail:

```
twisted.trial.unittest.FailTest: not equal:
a = 'twisted.scripts.tapconvert was deprecated in Twisted 12.1.0: Seek unzipping software outside of Twisted.'
b = 'twisted.scripts.tapconvert was deprecated in Twisted 12.1.0: tapconvert has been deprecated.'
```

(full build results [here](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-tapconvert-5038))

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [33560]) address review comments, refs [#5038](https://github.com/twisted/twisted/issues/5038)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner|
|-|-|

Replying to [jesstess](#note_5):
> Thanks for all your work cleaning up the cruft parts of the Twisted code base, thijs.
> 
> There's a copy-paste error from `test_deprecatedTkunzip` to `test_deprecatedTapconvert` that causes the new deprecation test to fail:

Ugh, sloppy, my bad, should be good now.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_10"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-tapconvert-5038) look good, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33580]) Merge deprecate-tapconvert-5038: The `tapconvert` utility is now deprecated.

Author: thijs
Reviewer: jesstess
Fixes: [#5038](https://github.com/twisted/twisted/issues/5038)
